packages/mdc-textfield/foundation.js

```diff
@@ -99,28 +146,45 @@ export default class MDCTextfieldFoundation extends MDCFoundation {
     this.adapter_.removeHelptextAttr(ARIA_HIDDEN);
   }
 
+  transitionEnd_(evt) {
+    const {BOTTOM_LINE_ACTIVE} = MDCTextfieldFoundation.cssClasses;
+
+    // We need to wait for the bottom line to entirely transparent
+    // before removing the class. If we do not, we see the line start to
+    // scale down before disappearing
+    if (evt.propertyName === 'opacity' && !this.isFocused_) {
+      this.adapter_.removeClassFromBottomLine(BOTTOM_LINE_ACTIVE);
+    }
+  }
+
   deactivateFocus_() {
-    const {FOCUSED, LABEL_FLOAT_ABOVE} = MDCTextfieldFoundation.cssClasses;
+    const {FOCUSED, LABEL_FLOAT_ABOVE, LABEL_SHAKE} = MDCTextfieldFoundation.cssClasses;
     const input = this.getNativeInput_();
 
     this.adapter_.removeClass(FOCUSED);
+    this.adapter_.removeClassFromLabel(LABEL_SHAKE);
+
     if (!input.value && !this.isBadInput_()) {
       this.adapter_.removeClassFromLabel(LABEL_FLOAT_ABOVE);
+      this.adapter_.removeClassFromLabel(LABEL_SHAKE);
```

Comment:
Why does this need to be here when it's already executed unconditionally 4 lines above?